### PR TITLE
Generate docs

### PR DESCRIPTION
**Pull request by pschoffer**
## Description

Generates JSON docs of the CLI and pushes it to the docs site.

Note: Currently using develop we should point it to prod once the WordPress parts get releasesd.

## Example result (for subset of commands only)

```
{
  "usage": "vip",
  "commands": [
    {
      "usage": "vip app [options] [command]",
      "commands": [
        {
          "usage": "vip app-list [options]",
          "name": "list",
          "description": "List your VIP applications"
        }
      ],
      "name": "app",
      "description": "List and modify your VIP applications"
    },
    {
      "usage": "vip cache [options] [command]",
      "commands": [
        {
          "usage": "vip cache list",
          "name": "list"
        }
      ],
      "name": "cache",
      "description": "Manage page cache for your VIP applications"
    }
  ],
  "options": [
    {
      "option": "-d, --debug",
      "description": "Activate debug output"
    },
    {
      "option": "-h, --help",
      "description": "Output the help for the (sub)command"
    },
    {
      "option": "-v, --version",
      "description": "Output the version number"
    }
  ],
  "version": "test42"
}
```

## Steps to Test

./helpers/generate-docs.js

### CODE REVIEW

File: helpers/generate-docs.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    for await (const data of childProcess.stdout) {
+        output += data.toString();
+    }
+    for await (const data of childProcess.stderr) {
+        output += data.toString();
+    }
+
+    const exitCode = await new Promise(resolve => {
+        childProcess.on('exit', resolve);
+    });
```
Comment (sjinks) [inline helpers/generate-docs.js]:
```js
const { once } = require('events');

const exitCode = await once(childProcess, 'exit');

```

File: helpers/generate-docs.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            continue;
+        }
+        if (line.startsWith('Examples:')) {
+            result.examples = [];
+            currentSection = SECTION_EXAMPLES;
+            continue;
+        }
+
+        if (currentSection === SECTION_COMMAND) {
+            const [_, command, description] = line.match(COMMAND_REGEXP);
```
Comment (sjinks) [inline helpers/generate-docs.js]:
```suggestion
            const [, command, description] = line.match(COMMAND_REGEXP);
```

File: helpers/generate-docs.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            result.commands.push({
+                command,
+                description,
+            });
+            continue;
+        }
+        if (currentSection === SECTION_OPTIONS) {
+            if (line.match(OPTION_REGEXP)) {
+                const [_, option, description] = line.match(OPTION_REGEXP);
```
Comment (sjinks) [inline helpers/generate-docs.js]:
```suggestion
                const [, option, description] = line.match(OPTION_REGEXP);
```

File: helpers/generate-docs.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+
+    return result;
+}
+
+const processCommand = async (subcommands) => {
+    const output = await runCommand(subcommands);
+    const parsedOutput = parseOutput(output);
+
+    const commandCount = parsedOutput.commands && parsedOutput.commands.length || 0;
```
Comment (sjinks) [inline helpers/generate-docs.js]:
```suggestion
    const commandCount = parsedOutput.commands?.length || 0;
```

File: helpers/generate-docs.js (hunk 1 above)
Comment (pschoffer) [reply helpers/generate-docs.js]: Thanks this is nice!

**Review by sjinks — APPROVED** [review]

**Pull request merged**